_projects/coala_language_server.md

```diff
@@ -42,48 +42,110 @@ a rapidly growing list of editor and language communities:
+langserver for most if not all the supported editors. This is not because of
+the server's inability, but rather the need to configure the front end of the
+editor plugin such as defining how to start the server, communicate with it etc.
 
-The project is divided into two parts:
+The project is divided into three parts:
 
-1. Offer a basic languager server.
+1. Offer a basic language server.
 2. Implement a plugin based on the language server.
+3. Optimize the coala entry point and language server.
 
 #### Features
 
 - Support linting for multiple programming languages.
-- Support 'textDocument/publishDiagnostics', 'textDocument/didChange' and 'textDocument/didSave' requests.
+- The current language server has implementations for:
+  - 'textDocument/didSave'
+  - 'textDocument/didChange'
+  - 'textDocument/publishDiagnostics'
+  and this project will improve and extend the support to at least
+  the following request types:
+  - 'textDocument/didOpen'
+  - 'textDocument/didClose'
+  - 'workspace/formatting'
+  - 'window/showMessageRequest'
 - (Advanced) Multiple plugins Support (e.g. Eclipse Che).
+- Some additional request types that are either out of the scope
+  of this project or are marginally redundant to build but which would
+  improve the user friendliness or functionality  of this language
+  server are:
+  - 'textDocument/willSaveWaitUntil'
+  - 'workspace/didChangeConfiguration'
+  - 'workspace/configuration'
+  - 'workspace/applyEdit'
+  - 'window/showMessage'
+
+#### Interface Optimizations
+
+The current implementation suffers from massive performance
+problems. This is primarily due to the mode in which lang
+server interacts with coala. Currently it works by executing
+coala via cli and redirecting stdout to language server. This
+is not an efficient mechanism because of all the involved overhead.
+This projects aims to provide a practical language server with
+performance improvements. Some ideas currently under consideration
+to achieve the said improvements are:
+
+  - Runtime coala configurations
+  - Language Server specific parameters in coafiles
+  - Custom entry point into coala
+
+Specific details about each method will be expanded upon
+once they are discussed with the community.
 
 #### Milestones
 
 ##### PREPARATION/BONDING
 
 - Design documentation about the language server exists.
-  - Architecture of the language server
-  - Mechanism to support multiple languages.
+  - Architecture of the language server.
+  - Mechanism to support multiple languages and requests.
+- Evaluation on various ways to optimize coala and language
+  server interface complete and a related cEP is submitted
+  for review.
 
 ##### CODING PHASE 1
 
 - A basic language server for coala exists.
   - Support linting for python.
-- Basic test suite exists.
-- Basic Documentation exists including asciinema showing the
+  - Supports primary request types:
+    - 'textDocument/didOpen'
```

Comment:
I'd say `textDocument/didChange` is also a primary request type.

Reply:
didChange is being deferred to coding phase 2 as coala isnt very well organised for running on very small changes. It only knows about files atm.

Maybe 'primary ' needs to be replaced with a more accurate term